src/modules/views/Folder/SharedDrives.tsx

```diff
@@ -1,42 +1,83 @@
-import React, { useState } from 'react'
+import React, { useCallback } from 'react'
+import { useNavigate } from 'react-router-dom'
 
-import { useQuery } from 'cozy-client'
-import { NavDesktopLimiter } from 'cozy-ui/transpiled/react/Nav'
-import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
-import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+import { useQuery, useClient } from 'cozy-client'
+import { IOCozyFile } from 'cozy-client/types/types'
+import flag from 'cozy-flags'
 
+import { SHARED_DRIVES_DIR_ID } from 'constants/config'
+import { useThumbnailSizeContext } from 'lib/ThumbnailSizeContext'
 import { CozyFile } from 'models/index'
 import { FileWithSelection as File } from 'modules/filelist/File'
 import { FileList } from 'modules/filelist/FileList'
 import FileListBody from 'modules/filelist/FileListBody'
-import { NavItem } from 'modules/navigation/NavItem'
+import { FileListHeader } from 'modules/filelist/FileListHeader'
+import { useFolderSort } from 'modules/navigation/duck'
+import generateShortcutUrl from 'modules/views/Folder/generateShortcutUrl'
 import { buildSharedDrivesQuery } from 'modules/views/Folder/queries/fetchSharedDrives'
 import { UseSharedDrivesQuery } from 'modules/views/Folder/types'
 
 interface SharedDrivesProps {
-  actions: unknown[]
-  handleFileOpen?: (path: string) => void
-  navigateToFolder: (path: string) => void
-  isFlatDomain: boolean
+  canSort: boolean
 }
 
 export const SharedDrives = ({
-  handleFileOpen,
-  navigateToFolder
+  canSort
 }: SharedDrivesProps): JSX.Element | null => {
-  const { isMobile } = useBreakpoints()
-  const sharedDrivesQuery = buildSharedDrivesQuery()
+  const { isBigThumbnail, toggleThumbnailSize } = useThumbnailSizeContext()
+  const [sortOrder, setSortOrder] = useFolderSort(SHARED_DRIVES_DIR_ID)
+  const client = useClient()
+  const navigate = useNavigate()
+  const changeSortOrder = useCallback(
+    (folderId_legacy, attribute: string, order: string) =>
+      setSortOrder({ attribute, order }),
+    [setSortOrder]
+  )
+
+  const sharedDrivesQuery = buildSharedDrivesQuery({
+    sortAttribute: sortOrder.attribute,
+    sortOrder: sortOrder.order
+  })
   const { data } = useQuery(
     sharedDrivesQuery.definition,
     sharedDrivesQuery.options
   ) as UseSharedDrivesQuery
-  const clickState = useState(null)
-  const { t } = useI18n()
+
+  const handleFolderOpen = (): void => {
+    // noop because we only display shortcuts
+  }
+  const handleFileOpen = ({ file }: { file: IOCozyFile }): void => {
+    if (
+      file.cozyMetadata?.createdByApp === 'nextcloud' &&
+      flag('drive.show-nextcloud-dev')
+    ) {
+      navigate(`/nextcloud/${file._id}`)
+    } else {
+      const { subdomain: subDomainType } = (client?.getInstanceOptions() ?? {
+        subdomain: 'flat'
+      }) as { subdomain: string }
```

Comment:
I'm not sure you should call `getInstanceOptions()` here.

As you are in a web app, you can access `client.instanceOptions` as it is already initialized from DOM here: https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/CozyClient.js#L1677-L1689

But even better, you can call `client.capabilities.flat_subdomains` that is initialized at the same place and directly contains the `boolean` you are looking for. Also `client.capabilities` is typed [here](https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/CozyClient.js#L188-L191) so you dont need to use any `as XXX` notation

Last but not least, you can also use the [useCapabilities](https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/hooks/useCapabilities.jsx) hook if needed (but it is not typed, we should fix that someday)